agent/auto-config/auto_config.go

```diff
@@ -54,26 +54,20 @@ var (
 // then we will need to add some locking here. I am deferring that for now
 // to help ease the review of this already large PR.
 type AutoConfig struct {
-	builderOpts        config.BuilderOpts
+	acConfig           Config
```

Comment:
This is a pattern I've used before that I think works well. By storing the "Config" on the struct we can easily see which fields of the struct are mutable state (all the other fields), and which ones are immutable or dependencies (the fields in acConfig).

`logger` and `certMonitor` could also be used from `acConfig`, but there were a lot more references to those 2, so I left them along for now to reduce the size of this diff.

Reply:
certMonitor will be going away soon. I am working on merging auto-encrypt and auto-config into one thing and the certificate monitoring will be integrated into the auto-config package.